sdk_contrib/aws_v1_support/src/test/java/io/opentelemetry/sdk/contrib/trace/aws/propagation/AWSXRayPropagatorTest.java

```diff
@@ -0,0 +1,278 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.contrib.trace.aws.propagation;
+
+import static com.google.common.truth.Truth.assertThat;
+import static io.opentelemetry.sdk.contrib.trace.aws.propagation.AWSXRayPropagator.TRACE_HEADER_KEY;
+
+import io.grpc.Context;
+import io.opentelemetry.context.propagation.HttpTextFormat;
+import io.opentelemetry.context.propagation.HttpTextFormat.Setter;
+import io.opentelemetry.internal.StringUtils;
+import io.opentelemetry.trace.DefaultSpan;
+import io.opentelemetry.trace.SpanContext;
+import io.opentelemetry.trace.SpanId;
+import io.opentelemetry.trace.TraceFlags;
+import io.opentelemetry.trace.TraceId;
+import io.opentelemetry.trace.TraceState;
+import io.opentelemetry.trace.TracingContextUtils;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class AWSXRayPropagatorTest {
+
+  private static final String TRACE_ID_FIRST_PART = "8a3c60f7";
+  private static final String TRACE_ID_SECOND_PART = "d188f8fa79d48a391a778fa6";
+  private static final String TRACE_ID_BASE16 = TRACE_ID_FIRST_PART + TRACE_ID_SECOND_PART;
+  private static final TraceId TRACE_ID = TraceId.fromLowerBase16(TRACE_ID_BASE16, 0);
+  private static final TraceState TRACE_STATE_DEFAULT = TraceState.builder().build();
+
+  private static final String SPAN_ID_BASE16 = "53995c3f42cd8ad8";
+  private static final SpanId SPAN_ID = SpanId.fromLowerBase16(SPAN_ID_BASE16, 0);
+  private static final TraceFlags SAMPLED_TRACE_FLAG = TraceFlags.builder().setIsSampled(true).build();
+
+  private static final HttpTextFormat.Setter<Map<String, String>> setter =
+      new HttpTextFormat.Setter<Map<String, String>>() {
+        @Override
+        public void set(Map<String, String> carrier, String key, String value) {
+          carrier.put(key, value);
+        }
+      };
+  private static final HttpTextFormat.Getter<Map<String, String>> getter =
+      new HttpTextFormat.Getter<Map<String, String>>() {
+        @Nullable
+        @Override
+        public String get(Map<String, String> carrier, String key) {
+          return carrier.get(key);
+        }
+      };
+  private final AWSXRayPropagator xrayPropagator = new AWSXRayPropagator();
+
+  @Test
+  public void inject_SampledContext() {
+    Map<String, String> carrier = new LinkedHashMap<>();
+    xrayPropagator.inject(
+        withSpanContext(
+            SpanContext.create(TRACE_ID, SPAN_ID, SAMPLED_TRACE_FLAG, TRACE_STATE_DEFAULT),
+            Context.current()),
+        carrier,
+        setter);
+
+    assertThat(carrier)
+        .containsEntry(
+            TRACE_HEADER_KEY,
+            generateTraceHeaderValue(
```

Comment:
I'd inline the strings, or have constants for the various forms. It's more readable and avoids the issue of now we need a test for this test helper.

Reply:
Will fix in rev2.